cmake/nanobind-config.cmake

```diff
@@ -5,41 +5,57 @@ if (NOT TARGET Python::Module)
 endif()
 
 # Determine the right suffix for ordinary and stable ABI extensions.
-# Python_SOSABI is guaranteed to be available in CMake 3.26+, and it may
-# also be available as part of backported FindPython in scikit-build-core
-if (DEFINED Python_SOSABI)
-  if (WIN32)
-    set(NB_SUFFIX_EXT ".pyd")
-  else()
-    set(NB_SUFFIX_EXT "${CMAKE_SHARED_MODULE_SUFFIX}")
-  endif()
 
-  set(NB_SUFFIX   ".${Python_SOABI}${NB_SUFFIX_EXT}")
+# We always need to know the extension
+if(WIN32)
+  set(NB_SUFFIX_EXT ".pyd")
+else()
+  set(NB_SUFFIX_EXT "${CMAKE_SHARED_MODULE_SUFFIX}")
+endif()
+
+# This was added in CMake 3.17+, also available earlier in scikit-build-core.
+# PyPy sets an invalid SOABI (platform missing), causing older FindPythons to
+# report an incorrect value. Only use it if it looks correct (X-X-X form).
+if(DEFINED Python_SOABI AND "${Python_SOABI}" MATCHES ".+-.+-.+")
+    set(NB_SUFFIX ".${Python_SOABI}${NB_SUFFIX_EXT}")
```

Comment:
super-minor: Is the indentation too deep here?

Reply:
Probably. I usually use `clang-format` in `pre-commit` so I don't have to think about things like that. :)

I'll run the downstream test first before fixes.

Reply:
```suggestion
  set(NB_SUFFIX ".${Python_SOABI}${NB_SUFFIX_EXT}")
```